ztp/source-crs-builder/MachineConfigChronyDynamicMaster/ptp-sync-check

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+# Evaluate PTP synchronization state and create or delete 
+# the chronyd.service condition file accordingly
+
+# The parametes below are tuned using chronyd-restart.service drop-in
+# Maximum PTP offset to be considered in-sync (ns)
+MAX_OFFSET=${MAX_OFFSET:-100000}
+# PTP offset hysteresis threshold (ns)
+OFFSET_THRESHOLD=${OFFSET_THRESHOLD:-20000}
+# Maximum valid measurement age (s)
+SYNC_AGE=${SYNC_AGE:-600}
+
+# File for starting chronyd conditionally
+SYNC_MARKER_PATH="/var/run/ptp/insync"
+
+# Auxiliary logging function
+function log_debug(){
+  local LOGGER_TAG="ptp-chronyd"
+  logger -t $LOGGER_TAG -p "syslog.debug" "$1"
+}
+
+# Compare floating point numbers
+function greater() {
+   awk -v n1="$1" -v n2="$2" 'BEGIN {printf (n1>n2?"true":"false")"\n", n1, n2}'
+}
+
+# Return absolute (for master offset)
+function abs() { 
+    echo ${1##-}
+}
+
+# Check variable is defined
+function var_defined() {
+  VAR="$1"
+  if [[ ${#VAR} -eq 0 ]]; then 
+    log_debug "Sanity check result at line $(caller): PTP might not be running"    
+    return 1
+  fi
+  return 0
+}
+
+# Get the last line containing masteroffset from the latest available log
+# ARGUMENTS:
+#   none
+# GLOBAL:
+#   none
+# OUTPUT:
+#   Last line of the log containing offset and timestamp
+#     Something like "2021-10-18T16:47:06.022983665+00:00 stdout F ptp4l[8180.150]: [ens4f0] 
+#     master offset 3 s2 freq -16756169 path delay 1481"
+# RETURN:
+#   0 if all sanity checks passed, 1 otherwise
+get_last_ptp_status() {
```

Comment:
With the switch to crictl, I think this is a lot clearer and a lot less fragile!  Nice!